modules/json/src/smithy4s/http/json/SchemaVisitorJCodec.scala

```diff
@@ -1215,7 +1218,10 @@ private[smithy4s] class SchemaVisitorJCodec(maxArity: Int)
             stage2 += {
               val value = buffer.get(f.label)
               if (f.isRequired) {
-                if (value == null) cursor.requiredFieldError(f.label, f.label)
+                if (value == null) {
+                  val label = jsonLabel(f)
+                  cursor.requiredFieldError(label, label)
```

Comment:
whilst we're at it, the first string here should probably be something else

Reply:
I noticed it but figured it was intended because it should be a `expected`, so I assume a type, but we were passing the label. My thinking was that a required field is not there so let's say we're `expected = fieldLabel` rather than `expected = int`, for example.